edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -117,11 +118,20 @@ protected internal async Task<T> Execute<T>(Func<Task<T>> func, string operation
             try
             {
                 Events.ExecutingOperation(operation, this.ManagementUri.ToString());
-                T result = await ExecuteWithRetry(func, r => Events.RetryingOperation(operation, this.ManagementUri.ToString(), r), this.transientErrorDetectionStrategy)
+                T result = await ExecuteWithRetry(
+                    func,
+                    (r) => Events.RetryingOperation(operation, this.ManagementUri.ToString(), r),
+                    this.transientErrorDetectionStrategy)
                     .TimeoutAfter(this.operationTimeout);
                 Events.SuccessfullyExecutedOperation(operation, this.ManagementUri.ToString());
                 return result;
             }
+            catch (SocketException ex) when (ex.Message.Contains("Connection refused"))
```

Comment:
We should also not use the exception message, since we are doing a rather drastic step.. so we need to be more careful

Reply:
Suggest to use SocketException.ErrorCode to compare a configurable value (int?) and add default value to appsettings json.